docs/architecture/grpc-for-wcf-developers/interface-definition-language.md

```diff
@@ -1,11 +1,29 @@
 ---
 title: Interface Definition Language - gRPC for WCF Developers
-description: TO BE WRITTEN
+description: Introducing the Protocol Buffers IDL.
 author: markrendle
 ms.date: 09/02/2019
 ---
 
 # Interface Definition Language
 
+With WCF, services can expose description metadata using the Web Service Definition Language (WSDL), which is generated dynamically using .NET Reflection at runtime. Developers can use this metadata to generate clients for those services, potentially in other languages if a platform-neutral binding such as SOAP over HTTP is used.
+
+gRPC uses the Interface Definition Language (IDL) from Protocol Buffers. The Protocol Buffers IDL is a custom, platform-neutral language with an open specification. Developers hand-code `.proto` files to describe services along with their inputs and outputs. These `.proto` files can then be used to generate language- or platform-specific stubs for clients and servers, allowing multiple different platforms to communicate. By sharing `.proto` files, teams can generate code to use each others' services without needing to take a code dependency.
+
+One of the advantages of the Protobuf IDL is that as a custom language it enables gRPC to be completely language and platform agnostic, not favoring any technology over another.
+
+The Protobuf IDL is also designed for humans to both read and write, whereas WSDL is intended as a machine-readable/writable format. Changing the WSDL of a WCF service typically requires making the changes to the service code itself, running the service and regenerating the WSDL file from the server. By contrast, with a `.proto` file, changes are simple to apply with a text editor, and automatically flow through the generated code. Visual Studio 2019 builds `.proto` files in the background when they are saved; when using other editors such as VS Code the changes will be applied when the project is built.
+
+When compared with XML, and particularly SOAP, messages encoded using Protobuf have many advantages. Protobuf messages can be up to 10x smaller than the same data serialized as SOAP XML, and encoding, decoding and transmitting them over a network can be up to 100x faster.
```

Comment:
@markrendle Can we remove references to actual numbers here? It's something we'd like to add back in later once we've measured it in a re-produce-able test. We're trying to avoid upsetting the applecart for folks already using WCF.